docs/content/src/learn/guides/file-integrations/file-integration-using-ftp/src/ftp_listener/ftp_listener.bal

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/log;
+import ballerina/config;
+import wso2/ftp;
+
+
+// Map to store processed file details
+map<int> fileMap = {};
+
+// Create FTP Listener
+// CODE-SEGMENT-BEGIN: segment_1
+listener ftp:Listener dataFileListener = new({
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+        basicAuth: {
+            username: config:getAsString("FTP_USERNAME"),
+            password: config:getAsString("FTP_PASSWORD")
+        }
+    },
+    path: config:getAsString("FTP_LISTENER_PATH"),
+    fileNamePattern: config:getAsString("FTP_FILE_NAME_PATTERN"),
+    pollingInterval: config:getAsInt("FTP_POLLING_INTERVAL")
+});
+// CODE-SEGMENT-END: segment_1
+
+// Configurations for FTP Client
+// CODE-SEGMENT-BEGIN: segment_3
+ftp:ClientEndpointConfig ftpConfig = {
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+        basicAuth: {
+            username: config:getAsString("FTP_USERNAME"),
+            password: config:getAsString("FTP_PASSWORD")
+        }
+    }
+};
+
+// Create FTP Client
+ftp:Client ftpClient = new(ftpConfig);
+// CODE-SEGMENT-END: segment_3
+
+// Service to be invoked on file addition/deletion on FTP server
+// CODE-SEGMENT-BEGIN: segment_2
+service dataFileService on dataFileListener {
+    resource function processDataFile(ftp:WatchEvent fileEvent) {
+
```

Comment:
Unnecessary empty line. Correct in all places please.

Reply:
Fixed in #623.